### PR TITLE
Data samples order

### PR DESCRIPTION
**Pull request by jmorel**
⚠️ In order for this PR to be merged, https://github.com/SubstraFoundation/substra-tools/pull/43 and https://github.com/SubstraFoundation/substra-backend/pull/344 will have to be merged first.

Main PR: https://github.com/SubstraFoundation/substra-tools/pull/43

Fixes and activates the tests on data samples order.

### CODE REVIEW

File: tests/test_data_samples_order.py
Comment (samlesu) [inline tests/test_data_samples_order.py:92]: If I understand well, this test has ~4% chance of emitting a false positive. I think this is too high, and could be decrease without costing too much. Maybe 8 data samples?

**Review by samlesu — COMMENTED** [review]
Except a small comment LGTM

File: tests/test_data_samples_order.py
Comment (jmorel) [reply tests/test_data_samples_order.py:92]: What do you think of replacing the _shuffle method by something like:
```python
def _shuffle(items):
    items = sorted(items)
    return [items[1], items[0]] + items[2:]
```

This way this can never be in alphabetical order or any kind of order.

Comment (samlesu) [reply tests/test_data_samples_order.py:92]: Seems like a good option.

Could you add a comment to the shuffle function to explain its goal (nothing more than rewording _This way this can never be in alphabetical order or any kind of order._) ?

Comment (jmorel) [reply tests/test_data_samples_order.py:92]: Definitely!

Comment (jmorel) [reply tests/test_data_samples_order.py:87]: Done

**Review by AurelienGasser — APPROVED** [review]
Nice work!

LGTM assuming those two tests pass.

**Comment by jmorel** [comment]
Like I said on the substra-backend PR, the tests are now passing! Could you update your reviews @samlesu @AurelienGasser  ?

**Review by AurelienGasser — APPROVED** [review]

**Review by samlesu — DISMISSED** [review]

**Comment by AurelienGasser** [comment]
@jmorel I restarted the travis builds, in order to be able to merge the PR. However the tests fail. Maybe the local backend needs some changes?

**Comment by jmorel** [comment]
@AurelienGasser I just launched the CI manually and it's using the 0.6.1 docker images instead of 0.7.0, I'll try again disabling the cache.

**Comment by AurelienGasser** [comment]
@jmorel any update on this?

**Comment by jmorel** [comment]
There were 2 issues:
* one with the publication of substra-tools that's been fixed (the 0.7.0 package is available on pypi)
* `make test-local` required an update in substra that wasn't merged and was bundled with other changes. I extracted these changes in a dedicated PR https://github.com/SubstraFoundation/substra/pull/244 that should fix the issue once merged.

**Comment by Kelvin-M** [comment]
I've also rebased your branch @jmorel

**Review by Kelvin-M — APPROVED** [review]

**Pull request merged**
